src/main/java/org/jabref/logic/net/URLDownload.java

```diff
@@ -81,6 +82,11 @@ public URLDownload(String source) throws MalformedURLException {
     public URLDownload(URL source) {
         this.source = source;
         this.addHeader("User-Agent", URLDownload.USER_AGENT);
+
+        // mEDRA requires Accept header attribute with desired content type
+        if (source.toString().contains(Medra.API_URL)) {
+            this.addHeader("Accept", Medra.CONTENT_TYPE_JSON);
```

Comment:
Why don't you use the `application/x-bibtex` header as for the other DOI fetcher well? That would make fetching a lot easier!

Reply:
I made a few tests but bibtext type seems to not work as exptected with mEDRA.
Here are the outputs of URLDonwnload.asString():

**mEDRA**
`@article{OLLETT, Andrew_2018, title={S?tav?hana and N?g?rjuna}, ISSN={2507-0347}, url={http://doi.org/10.2143/JIABS.41.0.3285748}, DOI={10.2143/JIABS.41.0.3285748}, number={0}, journal={Journal of the International Association of Buddhist Studies}, publisher={Peeters online journals}, author={OLLETT, Andrew}, year={2018}, pages={421–472} }`

**INSPIREHEP** (similarly for GoogleScholar):
```
@article{Maggiore:2017jjr,
author = "Maggiore, M. and Campo, D. and Antonini, P. and Lombardi, A. and Manzolaro, M. and Andrighetto, A. and Monetti, A. and Scarpa, D. and Esposito, J. and Silvestrin, L.",
title = "{SPES: A new cyclotron-based facility for research and applications with high-intensity beams}",
doi = "10.1142/S0217732317400107",
journal = "Mod. Phys. Lett. A",
volume = "32",
number = "17",
pages = "1740010",
year = "2017"
}
```

From what I understand, as a result, BibtexParser is not able to parse mEDRA string and returns no entries, while it works correctly in the other case.

Reply:
In general this works, but the specific authors of the citation contain umlauts which seem to be not correclty returned:
http://doi.org/10.2143/JIABS.41.0.3285748

Reply:
I performed several other tests with random mEDRA DOIs retrieved at [https://api.medra.org/doiprefixlist?doiPrefix=10.2143](https://api.medra.org/doiprefixlist?doiPrefix=10.2143) but only 1 over maybe 50 happened to be correctly parsed in bibtex type. Json type, instead, never failed and the only errors was due to the 500 error code returned by mEDRA.
At this point i believe the bibtex style returned by mEDRA could be the issue, but since this is my first contribution I'm really open to any other advice or suggestion you think i could work on to tackle this issue.

I'm committing some refactoring of the previous code along with WebFetchers class modified to show mEDRA agency in the drop down list of the Web Search pane.

Reply:
Thanks for the investigation! I think it's fine to use the json as workaround then.


From a users perspective, I don't want to know to whom a DOI actually belongs or to manually try another DOI fetcher.
It would be nice if there was some "magic" in the background. I just enter the DOI (no matter who "owns" it ) and I get my bibtex entry from it. 

Some suggestions: Implement the IDBasedFetcher interface, it's meant for fetchers which work based on an ID, e.g DOI, ISBN
The SearchBasedFetcher is meant for fetcher where you can enter a search term e.g. "abcdf" 
Scenario:
1. User enters doi (e.g. new entry from DOI)
2. If the DOI starts with medra prefix number => call medra fetcher and return result 
 else use the normal DOI results.


We have a similar setup for ISBNFetcher, we first check site a  and then site b in case the first didn't return anything.
For parsing DOIs we have the DOI.java class. 
1. Add a method like getPrefix to return the prefix of the DOI and then you could use that in the above scenario

I hope that makes it a bit more clear now. Feel free to ask if you have any further questions or problems.

Reply:
Thanks @Siedlerchr, I really appreciate your suggestions.
Here are some more checks I've performed.

I've figured out that the DoiFetcher.performSearchById was correctly retrieving the bibtex of mEDRA DOIs, even before I made any changes, but cannot parse it for the reason already mentioned. Thus, JabRef not retrieving mEDRA DOIs metadata was not the issue in #6602. 

Furthermore, I have not found an easy way of knowing the agency by prefix through mEDRA api without retrieving a list of DOIs registered for a particular prefix and then checking whether this is empty or not, so I don't think a getPrefix method would help. 
The most straightforward way seems to be calling [https://doi.org/doiRA/10.2143/10.2143/JCS.2.0.583382](https://doi.org/doiRA/10.2143/JCS.2.0.583382) ([DOI handbook - 5.2.1](https://www.doi.org/doi_handbook/5_Applications.html#5.2.1)) which in turn returns a single object array as follow:
 `[
  {
    "DOI": "10.2143/JCS.2.0.583382",
    "RA": "mEDRA"
  }
]`
I've then used this and created a getAgency method on DoiFetcher. This in turn would provide the condition to move processing toward mEDRA fetcher. 

Last, as suggested I have implemented IdBasedParserFetcher instead of SearchBasedParserFetcher and overridden IdBasedParserFetcher.performSearchByID within Medra class.

I hope my info are clear and if you think this could be a good way of proceeding I would update the PR. 
In this case I should remove SearchBasedParserFetcher from the open PR. Can I do delete it directly from the Files Changed view of this PR? Is it the correct way to do it?

Thank you so much.

Reply:
Ah, yes, then the DoiFetcher getAgency makes more sense.
Just delete/modify your changes locally and push your new commit with the changes, the PR will then be updated automatically.